app/src/androidTest/java/com/nervousfish/nervousfish/test/MainSteps.java

```diff
@@ -69,22 +71,17 @@ public void createDatabase() throws Exception {
     @After
     public void deleteDatabase() {
         final IDatabase database = NervousFish.getServiceLocator().getDatabase();
-
         database.deleteDatabase();
     }
 
     @Given("^I am viewing the main activity$")
     public void iAmViewingMainActivity() {
         final Intent intent = new Intent();
         this.mActivityRule.launchActivity(intent);
-        allowPermissionsIfNeeded();
       
         try {
             onView(withText(R.string.no)).perform(click());
-        } catch (NoMatchingViewException ignore) {
-        }
-
-        allowPermissionsIfNeeded();
+        } catch (NoMatchingViewException ignore) { /* If no popup is displayed, that is OK */ }
```

Comment:
We don't know if Bluetooth is enabled (or even available) when the tests are run, so we don't know if a popup shows up requesting to enable Bluetooth. If it is there we don't want to enable it (because the permission might not yet have been given), and otherwise its OK 👍 We don't know if Bluetooth is enabled (or even available) when the tests are run, so we don't know if a popup shows up requesting to enable Bluetooth. If it is there we don't want to enable it (because the permission might not yet have been given), and otherwise its OK 👍